Enable sequential fitting to be copied to other runs

sf1919 opened the pull request:
**Description of work.**
In sequential fitting in Muon Analysis GUI it is now possible to set the fit parameters for one run and copy it to all the other runs. The changes are kept if you switch to TF Symmetry mode.

**To test:**
1. Open Muon Analysis GUI and load ```MUSR00015189```
2. Open the Group Tab and click Analyse (plot/fit) for all the groups listed (should be 4 in total)
3. Open the Fit Tab, right-click on the empty fit table and select ```Add function```
4. Search for and select ```GausOsc```
5. Open the Sequential Fitting Tab and change the value of Sigma on line 3 to a value of your choice. This cell should be the only one that changes
6. Tick the checkbox ```Copy fit parameters to all```
7. Change the value of A on line 5 to a value of your choice. The value of A in each line of the table should change to your new value.
8. Untick the checkbox ```Copy fit parameters to all```
9. Change the value of A in any row to a value of your choice. Only that cell should change, the rest of the column should retain the value you updated to in step 7.

Fixes #31238 . 


<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
31238_add_radio_button

sf1919 committed:
31238_view_slot

sf1919 committed:
31238 working solution

sf1919 committed:
31238 tf_symmetry fix

sf1919 committed:
31238 Added documentation

Added a description of the work to the new features for release in 6.2.

sf1919 committed:
31238 add tests

Add new test for new method and add missing tests for handle_updated_fit_parameter_in_table in seq_fitting_tab_presenter

sf1919 committed:
Fixed doc formatting errors

Revised muon release notes to format new feature description correctly

robertapplin reviewed (changes requested):
This new feature is working well and thanks for adding those tests. I have a few minor suggestions in the code but it all looks very good

sf1919 committed:
31238 review changes

Cleaned up the code to reduce lines of code in response to review changes requested.

robertapplin reviewed (approved):
Thanks for those changes, looks good to go
